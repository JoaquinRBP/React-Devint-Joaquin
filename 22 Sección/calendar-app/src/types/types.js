export const types = {
    uiOpenModal: '[UI] Open Modal',
    uiCloseModal: '[UI] Close Modal',
    eventAddNew: '[event] Add New',
    eventSetActive: '[event] Set Active',
    eventClearActive: '[event] Clear Active',
    eventUpdated: '[event] Update',
    eventDeleted: '[event] Delete',
}