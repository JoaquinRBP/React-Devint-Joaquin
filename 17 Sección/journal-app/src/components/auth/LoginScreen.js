import React from 'react'
import { Link } from 'react-router-dom'
import {useForm} from '../../hooks/useForm';
import {useDispatch} from 'react-redux';
import validator from 'validator';
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth';
import { EvalError, removeError } from '../../actions/ui';
import { useSelector } from 'react-redux';
export const LoginScreen = () => {
    //useDispatch -- hook
    const dispatch =useDispatch();
    const [values, handleInputChange] = useForm({
        email:'joaquin@gmail.com',
        password:123456
    });
    const {loading} = useSelector( state => state.ui);
    const {email, password} = values;
    const handleLogin = (e) => {
        e.preventDefault();
        if(isFormValid()){
            dispatch(startLoginEmailPassword(email,password));
        }
    }
    const handleGoogleLogin = (e) => {
        e.preventDefault();
        dispatch(startGoogleLogin());
    }
    const isFormValid = () => {
        if(!validator.isEmail(email)){
            dispatch(EvalError('Email is not valid'));
            //console.log('');
            return false;
        }else if(password.length<5){
            dispatch(EvalError('Password is not valid'));
            //console.log('Password is not valid');
            return false;
        }
            dispatch(removeError());
        return true;
    }
return (
    <>
        <h3 className="auth__title">LoginScreen</h3>
        <form 
        className="animate__animated animate__fadeIn animate__faster"
        onSubmit={handleLogin}>
            <input 
                type="text"
                placeholder="Email"
                name="email"
                className="auth__input"
                value={email}
                onChange={handleInputChange}
            />
            <input 
                type="pasword"
                placeholder="Password"
                name="password"
                className="auth__input"
                autoComplete="off"
                value={password}
                onChange={handleInputChange}
            />
            <button 
                type="submit"
                className="btn btn-primary btn-block"
                disabled={loading}
            >
                Login
            </button>
            <div className="auth__social-networks">
                <p>Login with Social Networks</p>
                    <div 
                        className="google-btn"
                        onClick={handleGoogleLogin}
                    >
                    <div className="google-icon-wrapper">
                        <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                    </div>
                    <p className="btn-text">
                        <b>Sign in with google</b>
                    </p>
                    </div>
            </div>
            <Link className="link" to="/auth/register">
                Create a new account
            </Link>
        </form>
    </>
)
}
