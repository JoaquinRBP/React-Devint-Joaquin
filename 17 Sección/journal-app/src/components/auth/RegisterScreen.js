import React from 'react'
import { Link } from 'react-router-dom'

export const RegisterScreen = () => {
return (
    <>
        <h3 className="auth__title">LoginScreen</h3>
        <form>
            <input 
                type="text"
                placeholder="Name"
                name="name"
                className="auth__input"
            />
            <input 
                type="text"
                placeholder="Email"
                name="email"
                className="auth__input"
            />
            <input 
                type="pasword"
                placeholder="Password"
                name="password"
                className="auth__input"
                autoComplete="off"
            />
            <input 
                type="pasword"
                placeholder="Confirm"
                name="password2"
                className="auth__input"
                autoComplete="off"
            />
            <button 
                type="submit"
                className="btn btn-primary btn-block mb-5"
            >
                Register
            </button>
            <Link className="link mt-5" to="/auth/login">
                All Ready Register?
            </Link>
        </form>
    </>
)
}
