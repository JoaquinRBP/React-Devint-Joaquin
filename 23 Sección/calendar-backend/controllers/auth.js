const {response} = require('express');
const { validationResult} = require('express-validator');
const crearUsuario = (req,res = response) => {
    
    const {nombre, email, password} = req.body;
    // if(nombre.length <5) {
    //     return res.status(400).json({
    //         ok: false,
    //         msg:'El nombre debe de ser mayor o igual a 5 letras'
    //     })
    // }

    res.status(201).json(
        {
            ok: true,
            msg:'register',
            nombre,
            email,
            password
        }
        ); 
}
const loginUsuario = (req,res=response) => {
    const {email, password} = req.body
    
    
    res.json({
        ok: true,
        msg:'Login',
        email,
        password
    });
}
const revalidarToken = (req,res=response) => {
    res.json({
        ok: true,
        msg: 'renew'
    });
}
module.exports = {
    crearUsuario,
    loginUsuario,
    revalidadToken: revalidarToken,
};