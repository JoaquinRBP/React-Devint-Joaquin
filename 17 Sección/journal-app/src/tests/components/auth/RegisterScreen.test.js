import { mount } from "enzyme";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { RegisterScreen } from "../../../components/auth/RegisterScreen";
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { types } from "../../../types/types";
const middlewares = [thunk]
const mockStore = configureStore(middlewares);
const initState ={
    auth: {},
    notes:{
        active:{
            id: 'My5MpVrJrktWUIGtHeJV',
            title: 'Hola',
            body: 'Mundo'
        }
    },
    ui:{
        loading: false,
        msgError:null
    }
}
let store = mockStore(initState);
const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <RegisterScreen />
        </MemoryRouter>
    </Provider>
    )
describe('Pruebas sobre RegisterScreen', () => {
    // beforeEach(() => {
    //     store = mockStore(initState);
    //     jest.clearAllMocks();
    // })
    test('Deberia estar todo Ok', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de hacer el dispatch de la acciÃ³n respectiva', () => {
        const emailField = wrapper.find('input[name="email"]');
        emailField.simulate('change',{
            target: {
                value: '',
                name: 'email'
            }
        });
        wrapper.find('form').simulate('submit',{
            preventDefault(){}
        })
        const actions = store.getActions();
        expect(actions[0]).toEqual({
            type: types.uiSetError,
            payload: 'Email is not valid',
        })
    });
    test('Debe de mostrar la caja de alerta con un error', () => {
        //Sobrescribimos valores
        const initState ={
            auth: {},
            ui:{
                loading: false,
                msgError:'Email is not valid',
            }
        }
        const store = mockStore(initState);
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <RegisterScreen />
                </MemoryRouter>
            </Provider>
        );
        expect(wrapper.find('.auth__alert-error').exists()).toBe(true);
        expect(wrapper.find('.auth__alert-error').text().trim()).toBe(initState.ui.msgError);
    })
    
    
    
})
