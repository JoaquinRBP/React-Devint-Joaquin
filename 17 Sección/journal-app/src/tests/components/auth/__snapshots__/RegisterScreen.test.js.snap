// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre RegisterScreen Deberia estar todo Ok 1`] = `
Array [
  <h3
    className="auth__title"
  >
    LoginScreen
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Joaquin"
    />
    <input
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="joaquin@gmail.com"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="pasword"
      value="123456"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm"
      type="pasword"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block mb-5"
      disabled={false}
      type="submit"
    >
      Register
    </button>
    <a
      className="link mt-5"
      href="/auth/login"
      onClick={[Function]}
    >
      All Ready Register?
    </a>
  </form>,
]
`;
